File: src-tauri/src/lib/tauri_commands/ffprobe.rs
use crate::{
    domain::{AudioStream, Chapter, ContainerInfo, SubtitleStream, VideoStream},
    ffprobe,
};

#[tauri::command]
pub async fn get_container_info(
    app: tauri::AppHandle,
    video_path: &str,
) -> Result<ContainerInfo, String> {
    let mut ffprobe = ffprobe::FFPROBE::new(&app)?;
    ffprobe.get_container_info(video_path).await
}

#[tauri::command]
pub async fn get_video_streams(
    app: tauri::AppHandle,
    video_path: &str,
) -> Result<Vec<VideoStream>, String> {
    let mut ffprobe = ffprobe::FFPROBE::new(&app)?;
    ffprobe.get_video_streams(video_path).await
}

#[tauri::command]
pub async fn get_audio_streams(
    app: tauri::AppHandle,
    video_path: &str,
) -> Result<Vec<AudioStream>, String> {
    let mut ffprobe = ffprobe::FFPROBE::new(&app)?;
    ffprobe.get_audio_streams(video_path).await
}

#[tauri::command]
pub async fn get_subtitle_streams(
    app: tauri::AppHandle,
    video_path: &str,
) -> Result<Vec<SubtitleStream>, String> {
    let mut ffprobe = ffprobe::FFPROBE::new(&app)?;
    ffprobe.get_subtitle_streams(video_path).await
}

#[tauri::command]
pub async fn get_chapters(app: tauri::AppHandle, video_path: &str) -> Result<Vec<Chapter>, String> {
    let mut ffprobe = ffprobe::FFPROBE::new(&app)?;
    ffprobe.get_chapters(video_path).await
}

File: src/tauri/commands/ffprobe.ts
import { core } from '@tauri-apps/api'

import {
  AudioStream,
  Chapter,
  ContainerInfo,
  SubtitleStream,
  VideoInfo,
  VideoStream,
} from '@/types/compression'

export function getVideoBasicInfo(
  videoPath: string,
): Promise<VideoInfo | null> {
  return core.invoke('get_video_basic_info', { videoPath })
}

export function getContainerInfo(
  videoPath: string,
): Promise<ContainerInfo | null> {
  return core.invoke('get_container_info', { videoPath })
}

export function getVideoStreams(
  videoPath: string,
): Promise<VideoStream[] | null> {
  return core.invoke('get_video_streams', { videoPath })
}

export function getAudioStreams(
  videoPath: string,
): Promise<AudioStream[] | null> {
  return core.invoke('get_audio_streams', { videoPath })
}

export function getSubtitleStreams(
  videoPath: string,
): Promise<SubtitleStream[] | null> {
  return core.invoke('get_subtitle_streams', { videoPath })
}

export function getChapters(videoPath: string): Promise<Chapter[] | null> {
  return core.invoke('get_chapters', { videoPath })
}
